* sorry folks, but we need the config file(s) to be configurable
  (this patch applies on top of a stupid patch from gentoo-x86 that
  makes this one necessary)

--- john-1.7.3.1/src/params.h
+++ john-1.7.3.1/src/params.h
@@ -101,8 +101,12 @@
 /*
  * File names.
  */
+#ifndef CFG_FULL_NAME
 #define CFG_FULL_NAME			"/etc/john/john.conf"
+#endif
+#ifndef CFG_ALT_NAME
 #define CFG_ALT_NAME			"/etc/john/john.ini"
+#endif
 #if JOHN_SYSTEMWIDE
 #define CFG_PRIVATE_FULL_NAME		JOHN_PRIVATE_HOME "/john.conf"
 #define CFG_PRIVATE_ALT_NAME		JOHN_PRIVATE_HOME "/john.ini"
